RecoTracker/FinalTrackSelectors/interface/TfGraphDefWrapper.h

```diff
@@ -0,0 +1,16 @@
+#ifndef RecoTracker_FinalTrackSelectors_TfGraphDefWrapper_h
+#define RecoTracker_FinalTrackSelectors_TfGraphDefWrapper_h
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+class TfGraphDefWrapper {
+public:
+  TfGraphDefWrapper(tensorflow::Session*);
+  ~TfGraphDefWrapper() { tensorflow::closeSession(session_); };
```

Comment:
Given the destructor, it would be safer to make this class non-copyable and non-movable. In principle movability could be achieved, but that would require changes in the destructor. (currently an accidental copy of `TfGraphDefWrapper` could lead to premature closing of the Session)

Reply:
Could we have some suggestions to implement it? Maybe try something like ‘TfGraphDefWrapper & operator=(TfGraphDefWrapper&) = delete’ ?

Reply:
Thanks Matti for catching this!

@minxiyang per the rule of five https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-five, if you define a custom destructor, you should explicitly state what happens to the move and copy constructors&operators. In this case, they should be deleted:

```C++
  ~TfGraphDefWrapper() { tensorflow::closeSession(session_); };
  TfGraphDefWrapper(const TfGraphDefWrapper&) = delete;
  TfGraphDefWrapper& operator=(const TfGraphDefWrapper&) = delete;
  TfGraphDefWrapper(TfGraphDefWrapper&&) = delete;
  TfGraphDefWrapper& operator=(TfGraphDefWrapper&&) = delete;
```